client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,369 @@
+ * @param {Object} singleElementFromTasksSection the parsed Object from the RPAf Code  
+ * @param {String} robotId the id of the current robot / ssot 
+ * @returns attributeObject for a single attribute
+ */
+const buildSingleAttributeObject = (currentElement, singleElementFromTasksSection, robotId) => {
+    let { rpaTask } = singleElementFromTasksSection;
+    if (!rpaTask) rpaTask = 'no Task defined'
+
+    return {
+        activityId: currentElement.id,
+        rpaApplication: singleElementFromTasksSection.rpaApplication,
+        rpaTask,
+        robotId
+    }
+}
+
+/**
+ * @description builds the parameterObject for a single element
+ * @param {Object} singleAtrributeObject the attribute Object of the current activity
+ * @param {Object} singleElementFromTasksSection the parsed Object from the RPAf Code   
+ * @param {Array} taskAndApplicationCombinations all combinations of applications and tasks
+ * @returns parameterObject for a single attribute
+ */
+const buildSingleParameterObject = (singleAtrributeObject, singleElementFromTasksSection, taskAndApplicationCombinations) => {
+    const { rpaApplication, activityId, rpaTask, robotId } = singleAtrributeObject;
+    const singleParamArray = singleElementFromTasksSection.paramArray;
+
+    const combinationObject = taskAndApplicationCombinations.filter((singleCombinationObject) =>
+        (singleCombinationObject.Application === rpaApplication && singleCombinationObject.Task === rpaTask)
+    )[0];
+
+
+    const parameterArray = combinationObject.inputVars.map((singleInputVariable, index) => {
+        const singleParameterObject = {
+            index: singleInputVariable.index,
+            infoText: singleInputVariable.infoText,
+            isRequired: singleInputVariable.isRequired,
+            name: singleInputVariable.name,
+            type: singleInputVariable.type,
+            value: (singleParamArray[index].startsWith('%%') && singleParamArray[index].endsWith('%%') ? '' : singleParamArray[index]),
+            requireUserInput: (!!(singleParamArray[index].startsWith('%%') && singleParamArray[index].endsWith('%%'))),
+        };
+
+        return singleParameterObject;
+    })
+
+    return {
+        activityId,
+        rpaParameters: parameterArray,
+        robotId
+    };
+}
+
+/**
+ * @description build the elementsArray of the ssot
+ * @param {Array} robotCodeTaskSection robot code w/o empty lines as an array of Strings
+ * @param {Array} declaredApplications all declared Aplications from ***settings*** section as Strings 
+ * @param {String} robotId the id of the current robot / ssot 
+ * @returns elementsArray with all needed properties
+ */
+const getElementsArray = (robotCodeTaskSection, declaredApplications, robotId) => {
+    const elementsArray = [];
+    const attributeArray = [];
+    const parameterArray = [];
+
+    if (typeof robotCodeTaskSection === 'undefined' || typeof declaredApplications === 'undefined') return undefined;
```

Comment:
Is there any better way to check for them being undefined?

Reply:
Please check if you find a better alternative and add it at all occurrences

Reply:
I had already [researched](https://stackoverflow.com/questions/3390396/how-can-i-check-for-undefined-in-javascript) this and concluded that both your method with the double negation and a shorter option (`myVar === "undefined"`) are possible, but many sources put them behind the option I used. (The last option will return an error, if myVar is not declared)
That's why I had decided on this option. Tell me if we want to leave it like this or if there is a possibility that I had not considered.

Reply:
Then it's fine with me either way